third_party/intel/lib/TritonRaiseBlockPointer/TritonRaiseBlockPointer.cpp

```diff
@@ -201,21 +202,368 @@ struct TritonRaiseBlockPointer
+        newInitArgs.push_back(s);
+      }
+
+      for (auto [j, s] : llvm::enumerate(state.strides)) {
+        newInitArgs.push_back(s);
+      }
+
+      if (state.getRank() == 0) {
+        assert(state.scalar);
+        // for scalar pointers, the scalar contains the offset and is the only
+        // relevant state that could be updated by the loop.
+        newInitArgs.push_back(state.scalar);
+      }
+
+      // Note that we want the knownPtrs to be indexed by block arg, but we
+      // only have index for now. Also, the state we record is the init
+      // arg, but want to to use newly created block arg. These block args
+      // are not created yet. We will translate this mapping later.
+      knownPtrsTmp.push_back(std::make_pair(i, state));
+      levelToBlockArgIndex[level].insert(i);
+    }
+
+    // Create a new scf::ForOp that uses updated init args and same loop body
+    auto newOp = builder.create<scf::ForOp>(
+        op.getLoc(), op.getLowerBound(), op.getUpperBound(), op.getStep(),
+        newInitArgs,
+        [&](OpBuilder &b, Location loc, Value iv, ValueRange args) {
+          IRMapping cloneMap;
+          cloneMap.map(op.getInductionVar(), iv);
+          cloneMap.map(op.getInitArgs(), newInitArgs);
+          cloneMap.map(op.getRegionIterArgs(), args);
+
+          for (auto &bodyOp : op.getRegion().getOps()) {
+            b.clone(bodyOp, cloneMap);
+          }
+        });
+
+    // Convert the book-keeping data structure to use the correct key and value.
+    // Key is converted from init arg index to newly created block arg, and
+    // Value's PtrState fields are converted from init arg to newly created
+    // block arg
+    int cnt = op.getRegionIterArgs().size();
+    for (auto [i, state] : knownPtrsTmp) {
+      for (auto it = state.offsets.begin(); it != state.offsets.end(); it++) {
+        *it = newOp.getRegionIterArgs()[cnt];
+        cnt++;
+      }
+
+      for (auto it = state.strides.begin(); it != state.strides.end(); it++) {
+        *it = newOp.getRegionIterArgs()[cnt];
+        cnt++;
+      }
+
+      if (state.getRank() == 0) {
+        assert(state.scalar);
+        state.scalar = newOp.getRegionIterArgs()[cnt];
+        cnt++;
+      }
+
+      // Record the PtrState for this pointer
+      auto key = newOp.getRegionIterArgs()[i];
+      knownPtrs[key] = state;
+      initArgIndexMap[i] = state;
+
+      // For tensors of pointers, create a tt.make_block_ptr at the beginning of
+      // the loop body that correspond to this region iter arg. In case it is
+      // used by tt.load/tt.store in the loop body before pointer updates, this
+      // will make sure rewriteLoadOp/rewriteStoreOp can use the analysis
+      // result. E.g., given the following input (%tensor_of_ptr is a block
+      // arg): scf.for (%tensor_of_ptr) {
+      //   %data = tt.load %tensor_of_ptr
+      //   // more operations to update %tensor_of_ptr
+      // }
```

Comment:
I think this was messed up by formatting:
```suggestion
      // arg): 
      // scf.for (%tensor_of_ptr) {
      //   %data = tt.load %tensor_of_ptr
      //   // more operations to update %tensor_of_ptr
      // }
```